### PR TITLE
Prover: fix full prover over Sepolia issues

### PR DESCRIPTION
**Pull request by AlexandreBelling**
This fixes 2 issues:

- Assignment of the functional public inputs
- Constraints for the rolling-hash update
- Constraints in the pi-interconnection over the rollingHash as well

### Checklist

* [ ] I wrote new tests for my new core changes.
* [ ] I have successfully ran tests, style checker and build against my new changes locally.
* [ ] I have informed the team of any breaking changes if there are any.

### CODE REVIEW

File: prover/backend/execution/prove.go [hunk 1]
```diff
@@ -143,8 +143,11 @@ func mustProveAndPass(
 
 		if setupCfgChecksum != traces.Checksum() {
 			// This check is failing on prod but works locally.
-			// 	utils.Panic("traces checksum in the setup manifest does not match the one in the config")
-			logrus.Warnf("the setup checksum does not match the provided trace limits: provided=%++v", traces)
+			// @alex: since this is a setup-related constraint, it would likely be
```
Comment (AlexandreBelling) [inline prover/backend/execution/prove.go:146]: That one is a revert of a change we tried on main

File: prover/circuits/execution/circuit.go [hunk 2]
```diff
@@ -63,17 +64,23 @@ func assign(
...
+				},
 			},
-		},
-		PublicInput: new(big.Int).SetBytes(funcInputs.Sum(nil)),
-	}
+			PublicInput: new(big.Int).SetBytes(funcInputs.Sum(nil)),
+		}
+	)
+
+	res.FuncInputs.Assign(&funcInputs)
```
Comment (AlexandreBelling) [inline prover/circuits/execution/circuit.go:81]: @Tabaie as a FYI

File: prover/circuits/pi-interconnection/assign.go [hunk 3]
```diff
@@ -287,7 +287,7 @@ func (c *Compiled) Assign(r Request) (a Circuit, err error) {
 			// @alex: Not sure this check is a duplicate because we already check
 			// that the state root hash is well-propagated and this should be
 			// enough that the rolling hash update events are emitted in sequence.
-			if executionFPI.FirstRollingHashUpdateNumber != lastRollingHashNumber+1 {
+			if executionFPI.FirstRollingHashUpdateNumber >= lastRollingHashNumber {
```
Comment (AlexandreBelling) [inline prover/circuits/pi-interconnection/assign.go]: @Tabaie as a FYI

File: prover/zkevm/prover/publicInput/public_input.go [hunk 4]
```diff
@@ -109,8 +109,8 @@ func NewPublicInputZkEVM(comp *wizard.CompiledIOP, settings *Settings, ss *state
 				AbsLogNum:    getCol("loginfo.ABS_LOG_NUM"),
 				AbsLogNumMax: getCol("loginfo.ABS_LOG_NUM_MAX"),
 				Ct:           getCol("loginfo.CT"),
-				OutgoingHi:   getCol("loginfo.ADDR_HI"),
-				OutgoingLo:   getCol("loginfo.ADDR_LO"),
+				DataHi:       getCol("loginfo.DATA_HI"),
```
Comment (AlexandreBelling) [inline prover/zkevm/prover/publicInput/public_input.go:112]: @bogdanbear As a FYI. IIRC, I am the one who introduced that bug during the switch to go-corset. I changed the name of the field as I was confused by the variable name. I had thought "outgoing" meant outgoing address.

File: prover/circuits/pi-interconnection/assign.go (hunk 3 above)
Comment (AlexandreBelling) [reply prover/circuits/pi-interconnection/assign.go]: @thedarkjester confirmed that the rolling hash was updated in batch. So the consecutivity check is moot here.

Comment (Tabaie) [reply prover/circuits/pi-interconnection/assign.go]: 1. Shouldn't it be `executionFPI.FirstRollingHashUpdateNumber <= lastRollingHashNumber` instead then? The if checks for what _shouldn't_ happen iirc.
2. Is the change also reflected on the circuit side?

Comment (AlexandreBelling) [reply prover/circuits/pi-interconnection/assign.go]: Oh right, it has not been committed. Just addressed!

**Review by bogdanbear — APPROVED** [review]

**Review by Tabaie — APPROVED** [review]

**Pull request merged**